packages/scan/src/Scan.spec.ts

```diff
@@ -0,0 +1,265 @@
+      const ms = findArg<number>(args, 'number');
+      const timer = mockedImplementation?.(callback, ms);
+
+      jest.runAllTimers();
+
+      return timer as NodeJS.Timeout;
+    });
+};
+
+describe('Scan', () => {
+  const id = 'roMq1UVuhPKkndLERNKnA8';
+  const mockedScans = mock<Scans>();
+
+  let scan!: Scan;
+  let options!: ScanOptions;
+  let spiedOptions!: ScanOptions;
+
+  beforeEach(() => {
+    useFakeTimers();
+    options = { id, scans: instance(mockedScans) };
+    spiedOptions = spy(options);
+    scan = new Scan(options);
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.useRealTimers();
+    reset<Scans | ScanOptions>(mockedScans, spiedOptions);
+  });
+
+  describe('issues', () => {
+    it('should return an empty list', async () => {
+      when(mockedScans.listIssues(id)).thenResolve([]);
+      when(mockedScans.getScan(id)).thenResolve({ status: ScanStatus.RUNNING });
+
+      const result = await scan.issues();
+
+      expect(result).toEqual([]);
+      verify(mockedScans.listIssues(id)).once();
+    });
+
+    it('should return a cached result if scan is done', async () => {
+      when(mockedScans.listIssues(id))
+        .thenResolve([])
+        .thenResolve([
+          {
+            id: 'pDzxcEXQC8df1fcz1QwPf9',
+            order: 1,
+            details:
+              'Cross-site request forgery is a type of malicious website exploit.',
+            name: 'Database connection crashed',
+            severity: Severity.MEDIUM,
+            protocol: 'http',
+            remedy:
+              'The best way to protect against those kind of issues is making sure the Database resources are sufficient',
+            cvss: 'CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:L',
+            time: new Date(),
+            originalRequest: {
+              method: HttpMethod.GET,
+              url: 'https://brokencrystals.com/'
+            },
+            request: {
+              method: HttpMethod.GET,
+              url: 'https://brokencrystals.com/'
+            }
+          }
+        ]);
+      when(mockedScans.getScan(id)).thenResolve({ status: ScanStatus.DONE });
+
+      const result = await scan.issues();
+
+      expect(result).toEqual([]);
+    });
+  });
+
+  describe('status', () => {
+    it.each([
+      ScanStatus.DONE,
+      ScanStatus.STOPPED,
+      ScanStatus.DISRUPTED,
+      ScanStatus.FAILED
+    ])('should stop consuming on the %s status', async (status: ScanStatus) => {
+      const expected: ScanState = {
+        status
+      };
+      when(mockedScans.getScan(id)).thenResolve(expected);
+
+      const result: ScanState[] = [];
+
+      for await (const state of scan.status()) {
+        result.push(state);
+      }
+
+      verify(mockedScans.getScan(id)).once();
+      expect(result).toEqual([expected]);
+    });
+
+    it('should update a status multiple times util last value is consumed', async () => {
```

Comment:
"last" is not clear enough here.  what if  smth like "should not query the status after the final state reached" or " "should not query the status after done()"?